extensions/sql/policy/store/src/test/java/org/eclipse/dataspaceconnector/slq/policy/store/SqlPolicyStoreTest.java

```diff
@@ -0,0 +1,212 @@
+        var poolDataSource = new ConnectionPoolDataSource(connectionPool);
+        dataSourceRegistry.register(DATASOURCE_NAME, poolDataSource);
+        txManager.registerResource(new DataSourceResource(poolDataSource));
+        sqlPolicyStore = new SqlPolicyStore(dataSourceRegistry, DATASOURCE_NAME, transactionContext, new TypeManager(), new PostgressStatements());
+
+        try (var inputStream = getClass().getClassLoader().getResourceAsStream("schema.sql")) {
+            var schema = new String(Objects.requireNonNull(inputStream).readAllBytes(), StandardCharsets.UTF_8);
+            transactionContext.execute(() -> SqlQueryExecutor.executeQuery(connection, schema));
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        connectionPool.close();
+    }
+
+    @Test
+    @DisplayName("Save a single policy that not exists")
+    void save_notExisting() {
+        var policy = getDummyPolicy(getRandomId());
+
+        sqlPolicyStore.save(policy);
+
+        var policyFromDb = sqlPolicyStore.findById(policy.getUid());
+        assertThat(policyFromDb).isNotNull();
+        assertThat(policy.getUid()).isEqualTo(policyFromDb.getUid());
+    }
+
+    @Test
+    @DisplayName("Save a single a single policy that already exists")
+    void save_alreadyExists() {
+        var id = getRandomId();
+        var policy = getDummyPolicy(id);
+
+        sqlPolicyStore.save(policy);
+
+        assertThatThrownBy(() -> sqlPolicyStore.save(policy)).isInstanceOf(EdcPersistenceException.class);
+    }
+
+    @Test
+    @DisplayName("Find policy by ID that exists")
+    void findById_whenPresent() {
+        Policy policy =  getDummyPolicy(getRandomId());
+        sqlPolicyStore.save(policy);
+
+        var policyFromDb = sqlPolicyStore.findById(policy.getUid());
+
+        assertThat(policy.getUid()).isEqualTo(policyFromDb.getUid());
+    }
+
+    @Test
+    @DisplayName("Find policy by ID when not exists")
+    void findById_whenNonexistent() {
+        assertThat(sqlPolicyStore.findById("nonexistent")).isNull();
+    }
+
+    @Test
+    @DisplayName("Find all policies with limit and offset")
+    void findAll_withSpec() {
+        var limit = 20;
+
+        var definitionsExpected = getDummyPolicies(50);
+        definitionsExpected.forEach(sqlPolicyStore::save);
+
+        var spec = QuerySpec.Builder.newInstance()
+                .limit(limit)
+                .offset(20)
+                .build();
+
+        var policiesFromDB = sqlPolicyStore.findAll(spec).collect(Collectors.toList());
+
+        assertThat(policiesFromDB).hasSize(limit);
+    }
+
+    @Test
+    @DisplayName("Delete existing policy")
+    void deleteById_whenExists() {
+        var policy = getDummyPolicy(getRandomId());
+
+        sqlPolicyStore.save(policy);
+        assertThat(sqlPolicyStore.findById(policy.getUid()).getUid()).isEqualTo(policy.getUid());
```

Comment:
not really necessary, as it would verify the `findById` rather than the `delete`

Reply:
I have removed it